lib/sequent/core/event_publisher.rb

```diff
@@ -0,0 +1,57 @@
+module Sequent
+  module Core
+    #
+    # EventPublisher ensures that, for every thread, events will be published in the order in which they are meant to be published.
+    #
+    # This potentially introduces a wrinkle into your plans: You therefore should not split a "unit of work" across multiple threads.
+    #
+    # If you do not want this, you are free to implement your own version of EventPublisher and configure sequent to use it.
+    #
+    class EventPublisher
```

Comment:
I am thinking to split this in 2 concerns. This might help in understanding the concepts better and might be easier to extend with other strategies. This also allows us to support the current behavior and be backward compatible.

1) The Events queue with is basically `process_events`
2) The FIFO strategy" which is basically `events.each { |event| events_queue.push(event) }`. 

When someone want LIFO (or depth first) the event should be prepended to the Queue.

Reply:
I think we should hold off on implementing something like this until someone asks for it / needs it.